go/chat/commands/base.go

```diff
@@ -0,0 +1,90 @@
+package commands
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/keybase/client/go/chat/globals"
+	"github.com/keybase/client/go/chat/utils"
+	"github.com/keybase/client/go/protocol/chat1"
+	"github.com/keybase/client/go/protocol/gregor1"
+	"github.com/keybase/client/go/protocol/keybase1"
+)
+
+type baseCommand struct {
+	globals.Contextified
+	utils.DebugLabeler
+	name        string
+	aliases     []string
+	usage       string
+	description string
+}
+
+func newBaseCommand(g *globals.Context, name, usage, desc string, aliases ...string) *baseCommand {
+	return &baseCommand{
+		Contextified: globals.NewContextified(g),
+		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), fmt.Sprintf("Commands.%s", name), false),
+		name:         name,
+		usage:        usage,
+		aliases:      aliases,
+		description:  desc,
+	}
+}
+
+func (b *baseCommand) getConvByName(ctx context.Context, uid gregor1.UID, name string) (res chat1.ConversationLocal, err error) {
+	find := func(mt chat1.ConversationMembersType, name string, topicName *string) (conv chat1.ConversationLocal, err error) {
+		convs, err := b.G().ChatHelper.FindConversations(ctx, true, name, topicName,
+			chat1.TopicType_CHAT, mt, keybase1.TLFVisibility_PRIVATE)
+		if err != nil {
+			return res, err
+		}
+		if len(convs) == 0 {
+			return res, errors.New("conversation not found")
+		}
+		return convs[0], nil
+	}
+	if strings.Contains(name, "#") {
+		toks := strings.Split(name, "#")
+		return find(chat1.ConversationMembersType_TEAM, toks[0], &toks[1])
+	}
+	if res, err = find(chat1.ConversationMembersType_IMPTEAMNATIVE, name, nil); err != nil {
+		return find(chat1.ConversationMembersType_TEAM, name, nil)
+	}
+	return res, nil
+}
+
+func (b *baseCommand) commandAndMessage(text string) (cmd string, msg string, err error) {
```

Comment:
i think this should consume the first token (unused everywhere) and return the tokenized list for use by the command directly